libs/designsystem/src/lib/components/list/list-swipe-action.type.ts

```diff
@@ -4,11 +4,23 @@ export type ListSwipeDirection = 'right' | 'left';
 
 export type ListSwipeEnd = 'start' | 'end';
 
-export interface ListSwipeAction {
-  position: ListSwipeDirection;
-  title: ((item: any) => string) | string;
-  icon?: ((item: any) => string) | string;
-  type?: ((item: any) => ListSwipeActionType) | ListSwipeActionType;
+export type ItemSwipeAction = {
+  title: string;
+  // TODO: item is not provided in the function. Remove it.
   onSelected: (item: any) => void;
-  isDisabled?: ((item: any) => boolean) | boolean;
-}
+  icon?: string;
+  type?: ListSwipeActionType;
```

Comment:
```suggestion
  type?: ItemSwipeActionType;
```

Move ListSwipeActionType to `item-sliding.types.d.ts` and reexport it with a new name from here to not break existing functionality.